contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,307 @@
+  rules:
+    namespaces:
+      - key: team
+        operator: Exists
+    podSelector: {}    
+    tolerations:
+       match:
+        - key: "mykey"
+          operator: "Equal"
+          value: "value"
+          effects: "NoSchedule"
+        - key: "other_key"
+          operator: "Exists"
+          effect: "NoExecute"
+```
+
+
+### Priority classes
+
+Priority class usage can be regulated using fine-grain rules with `priorityClasseName` field.
+
+##### Allow
+
+In this example, we only allow the `critical-job` priority.
+
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  action: allow
+  rules:
+    namespaces:
+      - key: team
+        operator: Exists
+    podSelector: {}
+    priorityClasseNames:
+       match: "critical-priority"
+```
+
+
+### Node Selector
+
+The `nodeSelector` field makes it possible to specify what pods are allowed or denied based on their nodeSelectors.
+
+#### Examples
+
+##### Complete policy
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  action: deny   
+  rules:
+    namespaces:
+      - key: team
+        operator: Exists
+    podSelector: {}
+    nodeSelectors:
+      match:
+        - failure-domain.beta.kubernetes.io/region: ""
+        - disk: "ssd"
+        - disk: "premium"
+```
+
+
+In this example, pods cannot be scheduled if they have all of the following at the same time:
+-  `disk: ssd` nodeSelector
+-  `disk: hdd` nodeSelector
+-  `failure-domain.beta.kubernetes.io/region` nodeSelector with any value.
+
+
+### Affinity rules
+
+As anti-affinity rules are really time-consuming, we must be able to restrict their usage for each type (`nodeAffinities`, `podAffinities`, `podAntiAffinities`) a schedulingpolicy can list allowed/denied constraints (`requiredDuringSchedulingIgnoredDuringExecution`
+or `requiredDuringSchedulingIgnoredDuringExecution`).
+
+#### Examples
+
+##### Basic policy
+
+
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  action: allow   
+  rules:
+    namespaces:
+      - key: team
+        operator: Exists
+    podSelector: {}
+    nodeAffinities:
+      requiredDuringSchedulingIgnoredDuringExecution:
+         match:
+          - key: "authorized-region"
+            operator: "NotIn"
+            values: ["eu-2", "us-1"]
+          - key: "PCI-region"
+            operator: "Exists"
+            values: []          
+      preferredDuringSchedulingIgnoredDuringExecution:
+         match:
+          - key: "flavor"
+            operator: In
+            values: ["m1.small", "m1.medium"]
+
+```
+
+In this example, we allow pods that nodeAffinity to select nodes having `authorized-region` without `eu-1` or `us-1` values, or nodes having `PCI-region` label set. On those filtered nodes we require the pod to prefer nodes with the lowest compute capabilities (`m1.small` or `m1.medium`)
```

Comment:
s/that nodeAffinity/with nodeAffinity/ ?

The matching here should match against the nodeAffinity rules of pods. The description is vague. I guess we need to specify that a match happens when a Pod has:
1. All the "required" and "preferred" sections.
1. Each section has the same keys and the same operators.
1. Values must be the same or subset of those of the pod.

Reply:
True.